### PR TITLE
feat(csharp): add logs and logger factory

### PR DESCRIPTION
**Pull request by morganleroi**
## 🧭 What and Why

🎟 JIRA Ticket: DI-1731

### Changes included:
- Add logger factory extension.

### CODE REVIEW

File: templates/csharp/netcore_project.mustache [hunk 1]
```diff
@@ -38,26 +38,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    {{#useCompareNetObjects}}
-    <PackageReference Include="CompareNETObjects" Version="4.82.0" />
-    {{/useCompareNetObjects}}
-    {{^useGenericHost}}
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="7.0.0" />
```
Comment (millotp) [inline templates/csharp/netcore_project.mustache:41]: why not use the latest version ?

**Review by millotp — COMMENTED** [review]
very nice !

File: templates/csharp/netcore_project.mustache (hunk 1 above)
Comment (morganleroi) [reply templates/csharp/netcore_project.mustache:41]: Good question.

Comment (morganleroi) [reply templates/csharp/netcore_project.mustache:41]: IDK why the deps has been taken on this version. Will fix this ofc.

Comment (morganleroi) [reply templates/csharp/netcore_project.mustache]: Done

**Review by millotp — APPROVED** [review]
gg !

**Pull request merged**